presto-main/src/main/java/com/facebook/presto/operator/BlockEncodingBuffers.java

```diff
@@ -0,0 +1,363 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.DictionaryBlock;
+import com.facebook.presto.spi.block.LongArrayBlock;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import io.airlift.slice.SliceOutput;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.array.Arrays.ensureCapacity;
+import static com.facebook.presto.operator.UncheckedByteArrays.setByteUnchecked;
+import static com.facebook.presto.spi.MoreByteArrays.fill;
+import static com.google.common.base.Verify.verify;
+import static io.airlift.slice.SizeOf.SIZE_OF_BYTE;
+import static io.airlift.slice.SizeOf.sizeOf;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public abstract class BlockEncodingBuffers
```

Comment:
Note `BlockEncoding` is in `presto-spi`. Does `BlockEncodingBuffers` has to be in `presto-main`.

Even if it has to be in `presto-main`, I am wondering if `com.facebook.presto.operator` is a good package for it. What about the same `com.facebook.presto.spi.block` package?

Reply:
@wenleix Thanks for reviewing the PR! Regarding the place for BlockEncodingBuffers, @mbasmanova and I specifically moved them from presto-spi to presto-main. The reason was we didn't wan to pollute presto-spi and these objects contains state information(not stateless) and only used by the PartitionedOutputOperator. BlockEncoding  is stateless. Therefore we thought it's better to keep them in the same package of the operators. I'd love to hear more opinions on this.